libr/core/cconfig.c

```diff
@@ -2169,6 +2162,9 @@ static int cb_searchin(void *user, void *data) {
 		}
 		return false;
 	}
+	if (r_str_startswith (node->name, "anal")) {
+		core->anal->opt.noncode = strncmp (node->value, "io.maps.x", strlen ("io.maps.x")) != 0;
```

Comment:
this strncmp is exactly the same as r_str_startswith() and it work with other cases.. im not sure if this is the right way to fix this. because if i do anal.in=io.maps.x i expect r2 to analyze on all the maps with the exec bit set., which is exactly not the meaning of noncode

Reply:
> if i do anal.in=io.maps.x i expect r2 to analyze on all the maps with the exec bit set

yes , if **exec bit is set** we are setting `anal.noncode = false` which means to analyze only on exec sections , i don't see the confusion here ? ,